include/revng/Model/DefinedType.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Type.h"
+
+/* TUPLE-TREE-YAML
+name: DefinedType
+type: struct
+inherits: Type
+fields:
+  - name: Definition
+    reference:
+      pointeeType: TypeDefinition
+      rootType: Binary
+TUPLE-TREE-YAML */
+
+#include "revng/Model/Generated/Early/DefinedType.h"
+
+class model::DefinedType : public model::generated::DefinedType {
+public:
+  using generated::DefinedType::DefinedType;
+
+  static UpcastableType make(TypeOfDefinition Definition) {
+    return UpcastableType::make<DefinedType>(false, std::move(Definition));
+  }
+
+  static UpcastableType makeConst(TypeOfDefinition Definition) {
+    return UpcastableType::make<DefinedType>(true, std::move(Definition));
+  }
```

Comment:
Not sure how future-proof is to have a method like this in case we start adding extra attributes beyond `IsConst`, but for now it works.